app/services/authentication/authenticator.rb

```diff
@@ -130,6 +130,9 @@ def update_user(user)
       user.tap do |model|
         user.unlock_access! if user.access_locked?
         user.assign_attributes(provider.existing_user_data)
+        if user.email.blank? && provider.name == "twitter"
+          user.update_column(:email, provider.get_email)
```

Comment:
I'm using `update_column` here because it won't trigger a confirmation email. I'm assuming the email they have on Twitter is valid already, like we do with the sign up process.

Reply:
Can you switch to `update_columns(email: email)` which is a bit clearer? I wonder if there's a way to test this